src/features/posts.ts

```diff
@@ -0,0 +1,53 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostState = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadPosts = createAsyncThunk(
+  'posts/fetch',
+  async (userId: number) => {
+    const result = await getUserPosts(userId);
+
+    return result;
+  },
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      return { ...state, posts: [] };
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(loadPosts.pending, state => {
+      return { ...state, loaded: false };
+    });
+
+    builder.addCase(loadPosts.fulfilled, (state, actions) => {
+      return { ...state, items: actions.payload };
+    });
```

Comment:
In the `loadPosts.fulfilled` reducer, you are replacing the entire `items` array with the new posts. This might be what you want if you only ever load posts once. However, if you load posts multiple times and want to keep the previously loaded posts, you should append the new posts to the existing `items` array instead of replacing it.